File: Leetcode-Problems/Minimum Absolute Distance Between Mirror Pairs.cpp
class Solution
{
public:
    int rev(int temp)
    {
        int rev = 0;
        while (temp > 0)
        {
            rev = rev * 10 + (temp % 10);
            temp /= 10;
        }
        return rev;
    }

    int minMirrorPairDistance(vector<int> &nums)
    {
        unordered_map<int, int> mp;
        int ans = INT_MAX;
        for (int i = 0; i < nums.size(); i++)
        {
            if (mp.find(nums[i]) != mp.end())
            {
                ans = min(ans, i - mp[nums[i]]);
            }
            mp[rev(nums[i])] = i;
        }
        if (ans == INT_MAX)
            return -1;
        return ans;
    }
};

File: Leetcode-Problems/Minimum Absolute Distance Between Mirror Pairs.java
class Solution {
    public int minMirrorPairDistance(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        int ans = Integer.MAX_VALUE;
        for (int i = 0; i < nums.length; i++) {
            int rev = reverse(nums[i]);
            if (mp.containsKey(nums[i]))
                ans = Math.min(ans, Math.abs(i - mp.get(nums[i])));
            mp.put(rev, i);
        }
        return ans == Integer.MAX_VALUE ? -1 : ans;
    }

    public int reverse(int n) {
        int rev = 0;
        while (n > 0) {
            rev = rev * 10 + (n % 10);
            n /= 10;
        }
        return rev;
    }
}